e2e/mapimachines_test.go

```diff
@@ -0,0 +1,171 @@
+		if platform != configv1.AWSPlatformType {
+			Skip(fmt.Sprintf("Skipping tests on %s, this only support on aws", platform))
+		}
+
+		if !framework.IsMachineAPIMigrationEnabled(ctx, cl) {
+			Skip("Skipping, this feature is only supported on MachineAPIMigration enabled clusters")
+		}
+	})
+
+	var _ = Describe("Mapi machine creation E2E", Ordered, func() {
+		var newMachine *machinev1.Machine
+		// This cleanup runs after each test spec.
+		AfterEach(func() {
+			By("Cleaning up created machine")
+			// If a new machine was created in the test, delete it.
+			if newMachine != nil {
+				err := cl.Delete(context.Background(), newMachine)
+				Expect(client.IgnoreNotFound(err)).To(Succeed(), "Failed to delete the new machine")
+				// Wait for the machine to be fully deleted.
+				Eventually(func() error {
+					key := client.ObjectKey{Name: newMachine.Name, Namespace: newMachine.Namespace}
+					err := cl.Get(context.Background(), key, &machinev1.Machine{})
+					if apierrors.IsNotFound(err) {
+						return nil
+					}
+					return err
+				}, DefaultTimeout, DefaultInterval).Should(Succeed(), "New machine was not deleted")
+			}
+		})
+
+		It("should create a new machine from an existing one,when NO existing CAPI machine and authoritativeAPI is mapi", func() {
+			ctx := context.Background()
+			machineList := &machinev1.MachineList{}
+			By(fmt.Sprintf("Listing worker machines in namespace: %s", machineAPINamespace))
+			workerLabelSelector := client.MatchingLabels{RoleLabel: "worker"}
+			err := cl.List(ctx, machineList, client.InNamespace(machineAPINamespace), workerLabelSelector)
+			Expect(err).NotTo(HaveOccurred(), "Failed to list worker machines")
+			Expect(machineList.Items).NotTo(BeEmpty(), "No worker machines found in the namespace to use as a template")
+
+			var templateMachine *machinev1.Machine
+			foundMapiMachine := false
+			for i, m := range machineList.Items {
+				if m.Spec.AuthoritativeAPI == "MachineAPI" {
+					templateMachine = &machineList.Items[i]
+					foundMapiMachine = true
+					break
+				}
+			}
+			if !foundMapiMachine {
+				Skip("No machine found with AuthoritativeAPI set to MachineAPI")
+			}
+			// Define the new machine based on the template.
+			newMachine = &machinev1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					// Using a predictable name based on the testcase in polarion.
+					Name:      fmt.Sprintf("%s-test-%s", templateMachine.Name, "ocp-81829"),
+					Namespace: templateMachine.Namespace,
+					Labels:    templateMachine.Labels,
+				},
+				Spec: *templateMachine.Spec.DeepCopy(),
+			}
+			// Clear status and other instance-specific fields that should not be copied.
+			newMachine.Spec.ProviderID = nil
+			newMachine.ObjectMeta.Labels = nil
+			By(fmt.Sprintf("Creating a new machine in namespace: %s", newMachine.Namespace))
+			// Create the new machine object in the cluster.
+			err = cl.Create(ctx, newMachine)
+			Expect(err).NotTo(HaveOccurred(), "Failed to create new machine")
```

Comment:
Why not inline these?
```suggestion
			Expect(cl.Create(ctx, newMachine)).To(Succeed(), "Failed to create new machine")
```